test/compile_test/cstdfloat_cmath_incl_test.cpp

```diff
@@ -0,0 +1,92 @@
+//  Copyright Matt Borland 2021.
+//  Use, modification and distribution are subject to the
+//  Boost Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Basic sanity check that header
+// #includes all the files that it needs to.
+//
+#include <boost/math/cstdfloat/cstdfloat_cmath.hpp>
+//
+// Note this header includes no other headers, this is
+// important if this test is to be meaningful:
+//
+#include "test_compile_result.hpp"
+
+void compile_and_link_test()
+{
+    #ifdef __float128
```

Comment:
__float128 is a type not a macro?

Looking back at our docs on this, they look to be a bit confused, or indeed flat out wrong in places, but I *think* the thing to do is if BOOST_FLOAT128_C is defined, then `boost::float128` is available and can be tested as you have here.